polars/polars-io/src/csv_core/utils.rs

```diff
@@ -106,44 +117,52 @@ pub fn infer_file_schema<R: Read + Seek>(
     skip_rows: usize,
     comment_char: Option<u8>,
 ) -> Result<(Schema, usize)> {
-    let mut reader = BufReader::new(reader);
-    let mut line = String::new();
-    for _ in 0..skip_rows {
-        reader.read_line(&mut line)?;
-        line.clear()
-    }
-
     // We use lossy utf8 here because we don't want the schema inference to fail on utf8.
     // It may later.
     let encoding = CsvEncoding::LossyUtf8;
-    // set headers to false otherwise the csv crate, skips them.
-    let csv_reader = init_csv_reader(reader, false, delimiter, comment_char);
 
-    let mut records = csv_reader.into_byte_records();
-    let header_length;
+    let mut bytes = get_reader_bytes(reader)?;
+    bytes = skip_line_ending(skip_whitespace(skip_bom(&bytes)).0)
```

Comment:
I think we can assign to a variable that holds a reference to the slice. `&[u8]`. That saves a vec allocation.